core-kotlin-modules/core-kotlin-concurrency-2/src/test/java/com/baeldung/coroutine/CoroutineInteropUnitTest.java

```diff
@@ -0,0 +1,109 @@
+package com.baeldung.coroutine;
+
+import io.reactivex.rxjava3.core.Single;
+import kotlin.Result;
+import kotlin.coroutines.Continuation;
+import kotlin.coroutines.CoroutineContext;
+import kotlin.coroutines.EmptyCoroutineContext;
+import kotlinx.coroutines.BuildersKt;
+import kotlinx.coroutines.CoroutineScopeKt;
+import kotlinx.coroutines.CoroutineStart;
+import kotlinx.coroutines.future.FutureKt;
+import kotlinx.coroutines.reactor.MonoKt;
+import kotlinx.coroutines.rx3.RxSingleKt;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.baeldung.coroutine.JavaInteropKt.checkIn;
+import static com.baeldung.coroutine.JavaInteropKt.checkInAsync;
+import static com.baeldung.coroutine.JavaInteropKt.checkInClosed;
+
+public class CoroutineInteropUnitTest {
+
+    public static final String WELCOME = "Welcome";
+
+    static class CustomContinuation<T> implements Continuation<T> {
+        private final CompletableFuture<T> future;
+
+        public CustomContinuation(CompletableFuture<T> future) {
+            this.future = future;
+        }
+
+        @Override
+        public void resumeWith(@NotNull Object o) {
+            if (o instanceof Result.Failure)
+                future.completeExceptionally(((Result.Failure) o).exception);
+            else
+                future.complete((T) o);
+        }
+
+        @NotNull
+        @Override
+        public CoroutineContext getContext() {
+            return EmptyCoroutineContext.INSTANCE;
+        }
+    }
+
+    @Test
+    void givenCustomContinuation_whenCheckIn_getResult() throws ExecutionException, InterruptedException {
```

Comment:
Let's put the word 'then' in the test method names -- as in givenCustomContinuation_whenCheckIn_thenGetResult